File: wrappers_ubuntu/Javascript/harbor.js
const https = require("https");
const fs = require("fs");

const url = "https://raw.githubusercontent.com/TukangM/Harbor/main/harbor_ubuntu.sh";
const destination = "harbor_ubuntu.sh";

https.get(url, (response) => {
  if (response.statusCode !== 200) {
    console.error(`Failed to download file: ${response.statusMessage}`);
    return;
  }

  response.pipe(fs.createWriteStream(destination))
    .on("finish", () => {
      // Set executable permission on the downloaded file
      fs.chmodSync(destination, "755");

      // Run the downloaded file
      const childProcess = require("child_process");
      childProcess.execSync(`sh ${destination}`, { stdio: "inherit" });
    });
})
.on("error", (error) => {
  console.error(`Error downloading file: ${error}`);
});

File: wrappers_ubuntu/Python/harbor.py
import os
import urllib.request

def main():
    url = "https://raw.githubusercontent.com/TukangM/Harbor/main/harbor_ubuntu.sh"
    destination = "harbor_ubuntu.sh"

    try:
        # Download the script file
        urllib.request.urlretrieve(url, destination)

        # Set executable permission on the downloaded file
        os.chmod(destination, 0o755)

        # Run the downloaded file
        os.system(f"sh {destination}")
    except Exception as e:
        print(f"Error downloading or running script: {e}")

if __name__ == "__main__":
    main()